pkg/webhook/workspace/handler/immutable.go

```diff
@@ -0,0 +1,49 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/che-incubator/che-workspace-operator/pkg/apis/workspace/v1alpha1"
+	"github.com/che-incubator/che-workspace-operator/pkg/config"
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+func (h *WebhookHandler) HandleImmutable(ctx context.Context, req admission.Request) admission.Response {
+	oldObj := &unstructured.Unstructured{}
+	newObj := &unstructured.Unstructured{}
+	err := h.parse(req, oldObj, newObj)
+	if err != nil {
+		return admission.Denied(err.Error())
+	}
+	allowed, msg := h.handleImmutableObj(oldObj, newObj, req.UserInfo.UID)
+	if allowed {
+		return admission.Allowed(msg)
+	}
+	return admission.Denied(msg)
+}
+
+func (h *WebhookHandler) handleImmutableWorkspace(oldWksp, newWksp *v1alpha1.Workspace, uid string) admission.Response {
+	creatorUID := oldWksp.Labels[config.WorkspaceCreatorLabel]
+	if uid == creatorUID || uid == h.ControllerUID {
+		return admission.Allowed("immutable workspace is updated by owner or controller")
+	}
+	if cmp.Equal(oldWksp, newWksp, StopStartDiffOption[:]...) {
+		return admission.Allowed("immutable workspace is started/stopped")
+	}
+	return admission.Denied(fmt.Sprintf("workspace '%s' is immutable. To make modifications it must be deleted and recreated", oldWksp.Name))
```

Comment:
Seems the message should be rewritten, like `create your own workspace and then modify it` :stuck_out_tongue_closed_eyes:

Reply:
Good catch -- done :)